### ISSUE TITLE
Support Regex in module value, function in rename value when selecting `module`

### ISSUE THREAD

**Issue by tjcouch-sil** [UmamiAppearance/ImportManager#47]
Hi! 👋 
      
Firstly, thanks for your work on this project! 🙂

Today I used [patch-package](https://github.com/ds300/patch-package) to patch `rollup-plugin-import-manager@0.5.3` for the project I'm working on [source code for the plugin configuration here](https://github.com/paranext/paranext-extension-template/blob/94fd341d4d4c80e1fb41122ca425dfc2b317da19/vite/vite.config.ts#L39).

I needed to make exact matches to module names (instead of the current matching where it looks for the config-provided module name anywhere in the module name string) so that I could remove a specific module but not modules whose names included that module name, so I added Regex support in the module name.

I also needed to modify the relative path to modules programmatically when renaming them, not only replace module names one-for-one, so I added the ability to provide a function that receives the full raw module source path and returns a new full raw module source path.

Here is the diff that solved my problem:

```diff
diff --git a/node_modules/import-manager/README.md b/node_modules/import-manager/README.md
index 5754a5f..7451c36 100644
--- a/node_modules/import-manager/README.md
+++ b/node_modules/import-manager/README.md
@@ -49,7 +49,7 @@ Methods, callable from manager instance.
 Analyzes the source and stores all import statements as unit objects in the [imports object](#imports-object).
 
 ##### `selectModByName(name, type, allowNull)`
-Searches `this.imports` for the given module _name_. If _type_ is provided (`cjs`/`dynamic`/`es6`), it only searches for the module in that category. If _allowNull_ `false` the module must be found or a [`MatchError`](#matcherror
+Searches `this.imports` for the given module _name_ (`String` | `RegExp`). If _type_ is provided (`cjs`/`dynamic`/`es6`), it only searches for the module in that category. If _allowNull_ `false` the module must be found or a [`MatchError`](#matcherror
 ) is thrown.
 
 ##### `selectModById(id, allowNull)`
@@ -100,6 +100,8 @@ Methods callable from a unit object.
 ##### `renameModule(name, modType)`
 Changes the _name_ -> module (path). _modType_ can be `"string"` which adds quotation marks around _name_ or `"raw"`, which doesn't and can be used to pass variables if valid for the import type.
 
+EDIT: name can alternatively be (moduleSourceRaw: string) => string where moduleSourceRaw is the module's source code including quotes if _modType_ is `"string"`.
+
 ##### `addDefaultMembers(names)`
 _names_ is an array of strings (even for the most common case of a single member) of default members to add to the unit. _[es6 only]_
 
diff --git a/node_modules/import-manager/cjs/import-manager.cjs b/node_modules/import-manager/cjs/import-manager.cjs
index 7e54a1e..c6766a3 100644
--- a/node_modules/import-manager/cjs/import-manager.cjs
+++ b/node_modules/import-manager/cjs/import-manager.cjs
~~~ FILE REGENERATED ~~~
diff --git a/node_modules/import-manager/cjs/import-manager.cjs.map b/node_modules/import-manager/cjs/import-manager.cjs.map
index 7c48792..74fad9d 100644
--- a/node_modules/import-manager/cjs/import-manager.cjs.map
+++ b/node_modules/import-manager/cjs/import-manager.cjs.map
~~~ FILE REGENERATED ~~~
diff --git a/node_modules/import-manager/src/core.js b/node_modules/import-manager/src/core.js
index 28e1703..c806ecd 100644
--- a/node_modules/import-manager/src/core.js
+++ b/node_modules/import-manager/src/core.js
@@ -335,7 +335,9 @@ class ImportManager {
             start: node.source.start - node.start,
             end: node.source.end - node.start,
             type: "string",
-            quotes: node.source.raw.at(0)
+            quotes: node.source.raw.at(0),
+            // EDIT: added module's raw source code
+            sourceRaw: node.source.raw
         };
 
         
@@ -367,7 +369,9 @@ class ImportManager {
         const module = {
             name: importObject.source.value.split("/").at(-1) || "N/A",
             start: importObject.source.start - node.start,
-            end: importObject.source.end - node.start
+            end: importObject.source.end - node.start,
+            // EDIT: added module's raw source code
+            sourceRaw: importObject.source.raw
         };
 
         if (importObject.source.type === "Literal") {
@@ -403,7 +407,9 @@ class ImportManager {
         const module = {
             name: modulePart.value.split("/").at(-1) || "N/A",
             start: modulePart.start - node.start,
-            end: modulePart.end - node.start
+            end: modulePart.end - node.start,
+            // EDIT: added module's raw source code
+            sourceRaw: modulePart.raw
         };
 
         if (modulePart.type === "Literal") {
@@ -464,7 +470,7 @@ class ImportManager {
     
     /**
      * Selects a unit by its module name.
-     * @param {string} name - Module Name. 
+     * @param {string|RegExp} name - Module Name. 
      * @param {string|string[]} [type] - "cjs", "dynamic", "es6" one as a string or multiple as array of strings
      * @param {boolean} allowNull - If false the module must be found or a MatchError is thrown.
      * @returns {Object} - An explicit unit.
@@ -502,7 +508,8 @@ class ImportManager {
 
         // filter for unit name
         const units = unitList.filter(unit => {
-            const match = unit.module.name.indexOf(name) > -1;
+            // EDIT: unit.module can now be a string or a regex
+            const match = name instanceof RegExp ? unit.module.name !== undefined && name.test(unit.module.name) : unit.module.name.indexOf(name) > -1;
 
             // ignore deleted units
             if (match && unit.module.name.match(/^\(deleted\)/)) {
diff --git a/node_modules/import-manager/src/unit-methods.js b/node_modules/import-manager/src/unit-methods.js
index 26534df..1a50d0d 100644
--- a/node_modules/import-manager/src/unit-methods.js
+++ b/node_modules/import-manager/src/unit-methods.js
@@ -48,21 +48,27 @@ export default class ImportManagerUnitMethods {
 
     /**
      * Changes the module part of a import statement.
-     * @param {string} name - The new module part/path.
+     * @param {string | (moduleSourceRaw: string) => string} name - The new module part/path or a function
+     * that receives the module's raw source code (including quotes if present) and returns the new module part/path.
      * @param {*} modType - Module type (literal|raw).
+     * EDIT: Added support for name (aka rename in rollup plugin action) to be a function
      */
     renameModule(name, modType) {
-        if (modType === "string") {
-            if (!this.unit.module.quotes) {
-                this.unit.module.quotes = "\"";
+        const isNameAFn = typeof name === 'function';
+
+        if (!isNameAFn) {
+            if (modType === "string") {
+                if (!this.unit.module.quotes) {
+                    this.unit.module.quotes = "\"";
+                }
+                const q = this.unit.module.quotes;
+                name = q + name + q;
+            } else if (modType !== "raw") {
+                throw new TypeError(`Unknown modType '${modType}'. Valid types are 'string' and 'raw'.`);
             }
-            const q = this.unit.module.quotes;
-            name = q + name + q;
-        } else if (modType !== "raw") {
-            throw new TypeError(`Unknown modType '${modType}'. Valid types are 'string' and 'raw'.`);
         }
         
-        this.unit.code.overwrite(this.unit.module.start, this.unit.module.end, name);
+        this.unit.code.overwrite(this.unit.module.start, this.unit.module.end, isNameAFn ? name(this.unit.module.sourceRaw) : name);
 
         if (this.unit.type === "es6") {
             this.updateUnit();
```

This change corresponds to https://github.com/UmamiAppearance/rollup-plugin-import-manager/issues/54

Please let me know if you would like me to submit a pull request with these changes.

<em>This issue body was [partially generated by patch-package](https://github.com/ds300/patch-package/issues/296).</em>

**Comment by UmamiAppearance** [comment]
See https://github.com/UmamiAppearance/rollup-plugin-import-manager/issues/54

A new branch regarding this issue can be found here:
[regex-module-selection](https://github.com/UmamiAppearance/ImportManager/tree/regex-module-selection)

**Comment by tjcouch-sil** [comment]
Seems I don't have write permission to that branch, so I forked it and made a [PR into main here](https://github.com/UmamiAppearance/ImportManager/pull/48). Thank you!

**Comment by UmamiAppearance** [comment]
resolved in #49
